server/server.go

```diff
@@ -260,16 +264,20 @@ func (server *Server) startHTTPServers() {
 }
 
 func (server *Server) setupServerEntryPoint(newServerEntryPointName string, newServerEntryPoint *serverEntryPoint) *serverEntryPoint {
-	serverMiddlewares := []negroni.Handler{middlewares.NegroniRecoverHandler(), stats}
+	serverMiddlewares := []negroni.Handler{middlewares.NegroniRecoverHandler()}
 	if server.accessLoggerMiddleware != nil {
 		serverMiddlewares = append(serverMiddlewares, server.accessLoggerMiddleware)
 	}
 	if server.metricsRegistry.IsEnabled() {
 		serverMiddlewares = append(serverMiddlewares, middlewares.NewMetricsWrapper(server.metricsRegistry, newServerEntryPointName))
 	}
-	if server.globalConfiguration.Web != nil && server.globalConfiguration.Web.Statistics != nil {
-		statsRecorder = middlewares.NewStatsRecorder(server.globalConfiguration.Web.Statistics.RecentErrors)
-		serverMiddlewares = append(serverMiddlewares, statsRecorder)
+	if server.globalConfiguration.Web != nil {
```

Comment:
I wonder if these changes are compliant with the metrics refactor made in https://github.com/containous/traefik/pull/1968

Reply:
AFAICS it doesn't interfere with the metrics implementation but the naming confuses at the moment. Its a good step by the way to get rid of the global `stats` variable. Anyhow `MetricsRecorder` is a confusing naming. I would stick with `Stats`.